pkg/apis/serving/v1alpha1/service_defaults_test.go

```diff
@@ -264,6 +275,53 @@ func TestServiceDefaulting(t *testing.T) {
 				},
 			},
 		},
+	}, {
+		name: "inline defaults to run latest",
+		in: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{
+						Spec: RevisionSpec{
+							RevisionSpec: v1beta1.RevisionSpec{
+								PodSpec: corev1.PodSpec{
+									Containers: []corev1.Container{{
+										Image: "blah",
+									}},
+								},
+							},
+						},
+					},
+				},
+				// No RouteSpec should get defaulted to "run latest"
+			},
+		},
+		want: &Service{
+			Spec: ServiceSpec{
+				ConfigurationSpec: ConfigurationSpec{
+					RevisionTemplate: &RevisionTemplateSpec{
+						Spec: RevisionSpec{
+							RevisionSpec: v1beta1.RevisionSpec{
+								TimeoutSeconds: ptr.Int64(config.DefaultRevisionTimeoutSeconds),
+								PodSpec: corev1.PodSpec{
+									Containers: []corev1.Container{{
+										Image:     "blah",
+										Resources: defaultResources,
+									}},
+								},
+							},
+						},
+					},
+				},
+				RouteSpec: RouteSpec{
+					Traffic: []TrafficTarget{{
+						TrafficTarget: v1beta1.TrafficTarget{
+							LatestRevision: ptr.Bool(true),
+							Percent:        100,
+						},
+					}},
+				},
+			},
+		},
```

Comment:
Tests for:

Defaulting when Empty ConfigurationSpec is provided
Defaulting when Empty RevisionTemplate is provided
Defaulting for when empty traffic is provided (instead of nil)
Defaulting for when traffic with percent is provided, but no latestRevision

Reply:
"empty" is the first of those.

added the rest and then some.